zppy/templates/coupled_global.py

```diff
@@ -662,210 +1205,194 @@ def make_plot_pdfs(
+                    is_file=True,
+                    title=f"{rgn}_{component}_{plot_name}",
+                )
+
+    url = viewer.generate_page()
+    viewer.generate_viewer()
+    # Copy the contents of `table` into the `viewer` directory
+    # (which initially only has `css` and `js` subdirectories)
+    # Because `viewer` already exists,
+    # `shutil.copytree` will not work.
+    distutils.dir_util.copy_tree("table", "viewer")
+    print(
+        os.getcwd()
+    )  # /lcrc/group/e3sm/ac.forsyth2/zppy_min_case_global_time_series_viewers_output/test-pr616-20241022v2/v3.LR.historical_0051/post/scripts/global_time_series_1985-1995_dir
+    # shutil.rmtree("table")
+    # new_url = f"viewer_{component}"
+    # # shutil.move("viewer", new_url)
+    # distutils.dir_util.copy_tree("viewer", new_url)
+    return url
+
+
+# Copied from E3SM Diags and modified
+def create_viewer_index(
+    root_dir: str, title_and_url_list: List[Tuple[str, str]]
+) -> str:
+    """
+    Creates the index page in root_dir which
+    joins the individual viewers.
+
+    Each tuple is on its own row.
+    """
+
+    def insert_data_in_row(row_obj, name, url):
+        """
+        Given a row object, insert the name and url.
+        """
+        td = soup.new_tag("td")
+        a = soup.new_tag("a")
+        a["href"] = url
+        a.string = name
+        td.append(a)
+        row_obj.append(td)
+
+    install_path = ""  # TODO: figure this out
+    path = os.path.join(install_path, "viewer", "index_template.html")
+    output = os.path.join(root_dir, "index.html")
+
+    soup = BeautifulSoup(open(path), "lxml")
+
+    # If no one changes it, the template only has
+    # one element in the find command below.
+    table = soup.find_all("table", {"class": "table"})[0]
+
+    # Adding the title.
+    tr = soup.new_tag("tr")
+    th = soup.new_tag("th")
+    th.string = "Output Sets"
+    tr.append(th)
+
+    # Adding each of the rows.
+    for row in title_and_url_list:
+        tr = soup.new_tag("tr")
+
+        if isinstance(row, list):
+            for elt in row:
+                name, url = elt
+                insert_data_in_row(tr, name, url)
         else:
-            raise RuntimeError(f"Invalid rgn={rgn}")
-        run(parameters, rgn)
+            name, url = row
+            insert_data_in_row(tr, name, url)
+
+        table.append(tr)
+
+    html = soup.prettify("utf-8")
+
+    with open(output, "wb") as f:
+        f.write(html)
+
+    return output
+
+
+def run_by_region(command_line_arguments):
+    parameters: Parameters = Parameters(command_line_arguments)
+    requested_variables = RequestedVariables(parameters)
+    for rgn in parameters.regions:
+        run(parameters, requested_variables, rgn)
+    plots_per_page = parameters.nrows * parameters.ncols
+    # TODO: Is this how we want to determine when to make a viewer or should we have a `make_viewer` parameter in the cfg?
```

Comment:
Design decision question: how do we want the user to specify they want a Viewer?

- Always make a viewer
- Make a viewer if they say they want 1 row, 1 column summary pages (i.e., single plot pages) and otherwise construct the `nrows x ncols` summary PDFs as before?
- Have an explicit parameter in the `cfg`: `make_viewer = True`